doc/wg/core/notes/core-notes-2021-07-23.md

```diff
@@ -0,0 +1,178 @@
+ - Pat: What we have done before is a `Post-2.0` tag. Everything will either be merged, closed, or explicitly schosen to be deferred.
+ - Alexandru: I will fix in the next few days, should be ready by early next week.
+
+
+## Time HIL (https://github.com/tock/tock/pull/2665)
+ - Branden: Start with Jett for background?
+ - Jett: We have downstream code, and was using `Alarm` as a trait object. We were doing this for code cleanliness rather than size. Couldn't do that when updating since there are methods now. Doesn't seem like it was an explicit choice to prevent trait objects. The PR just moves this out to a separate trait that's auto-implemented. This now allows `Time` `Alarm` etc to be trait objects.
+ - Jett: Maybe for Alarm or Timer generic parameters will be correct, but we have seen some examples where dynamic trait object is better because otherwise there are too many copies. The goal here is just to allow the option of a trait object.
+ - Jett: The other change is it adds the corrolary `to` for the `from`s
+ - Branden: So the behavior didn't change, there is just some additional functionality, but how you can access it changed?
+ - Jett: Yeah, that's a good question. The end user can now use `Time` `Alarm` etc as a dynamic object if you want to. You have access to more methods. And you now access via `self.method()` rather than `A::method()`. You use the instance to do things rather than the trait.
+ - Jett: Hudson did code size analysis already, it's none/trivial.
+ - Leon: Apart from the formal issues of whether this requires a new TRD, I like the new syntax a lot, and it makes the code more readable. However, do want to express concern for dynamic trait objects, particularly in the time HIL, trait objects can significantly reduce the compiler's ability to make optimizations (which is harder with time as it's so pervasive).
+ - Leon: I also think that the compiler sometimes cannot correctly drop implementations. Bascially once people use it in the wrong way, size could blow up.
+ - Hudson: The flip side is that people are limited in what they can do.
+ - Leon: But that hasn't happened yet has it?
+ - Pat: My understanding was the downstream had exactly this need, which motivated this PR?
+ - Jett: That is correct.
+ - Loen: Ah, okay.
+ - Jett/Leon: In short, all of the upstream code should not be using it as a trait object since it doesn't make sense in their use cases.
+ - Alexandru: I started arount Tock 1.4 and had issues using Alarm as generic
+ - Hudson: Jett started at 1.6 and was able to use it as generic
+ - Phil: The underlying question is do you want to be able to check things at runtime or compile time? The intention was to be able to do all of this statically
+ - Loen: This confused me at first as well, but I don't think that doing this as a dynamic trait object loses this. I think because of all of the associated type specifications you still get the compile-time checks
```

Comment:
typo -- although also a good name :smile: 

```suggestion
 - Leon: This confused me at first as well, but I don't think that doing this as a dynamic trait object loses this. I think because of all of the associated type specifications you still get the compile-time checks
```